apps/prairielearn/src/pages/userSettings/userSettings.html.ts

```diff
@@ -192,6 +222,64 @@ export function UserSettings({
   `.toString();
 }
 
+function PurchaseTable({ purchases }: { purchases: PurchaseRow[] }) {
+  return html`
+    <div class="table-responsive">
+      <table class="table">
+        <thead>
+          <tr>
+            <th>ID</th>
+            <th>Course</th>
+            <th>Date</th>
+            <th>Amount</th>
+            <th>Status</th>
+          </tr>
+        </thead>
+        <tbody>
+          ${purchases.map((purchase) => {
+            const courseName = purchase.course
+              ? `${purchase.course.short_name}: ${purchase.course.title}`
+              : 'Unknown course';
+
+            const courseInstanceName =
+              purchase.course_instance?.long_name ?? 'Unknown course instance';
```

Comment:
We *should* always have a corresponding course and course instance to display since we soft-delete them, but I wanted to be extra careful since we're doing `LEFT JOIN`s and our schema doesn't enforce a non-null `course_instance_id`.

Writing this out: I think to be extra safe and not lose data, we should consider removing `ON DELETE CASCADE` from at least `stripe_checkout_sessions.course_instance_id`...

Reply:
I think we should remove all the foreign key constraints from `stripe_checkout_sessions`. We are using this as a log/audit table so we don't want to delete rows from it under any circumstances?

Reply:
That's a good point - let me do that.

Reply:
Done in 95c6551.